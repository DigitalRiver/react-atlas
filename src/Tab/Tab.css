@import '../styles/variables.css';

.tab {
  composes: bg-silver from '../styles.css';
  composes: sky-blue from '../styles.css';
  composes: border-colorless from '../styles.css';
  composes: border-transparent from '../styles.css';
  composes: border-bottom-none from '../styles.css';
  composes: marg-r-half from '../styles.css';
  composes: pad-v-1 from '../styles.css';
  composes: pad-h-2 from '../styles.css';
  composes: cursor-pointer from '../styles.css';
  border-top: 1px solid var(--white);
  display: inline-block;
  position: relative;
  list-style: none;
}

.vTab {
  composes: bg-silver from '../styles.css';
  composes: sky-blue from '../styles.css';
  composes: border-colorless from '../styles.css';
  composes: border-transparent from '../styles.css';
  composes: border-right-none from '../styles.css';
  composes: marg-b-half from '../styles.css';
  composes: pad-v-2 from '../styles.css';
  composes: pad-h-1 from '../styles.css';
  composes: cursor-pointer from '../styles.css';
  border-left: 1px solid var(--white);
  display: block;
  position: relative;
  list-style: none;
}

.selected {
  composes: bg-white from '../styles.css';
  composes: dark-gray from '../styles.css';
  border-color: var(--gray) !important; /* Important designation added for production build */
  bottom: -1px;
}

.vSelected {
  composes: bg-white from '../styles.css';
  composes: dark-gray from '../styles.css';
  border-color: var(--gray) !important; /* Important designation added for production build */
  right: -1px;
}

.disabled {
  composes: bg-white from '../styles.css';
  composes: cursor-default from '../styles.css';
  composes: dark-gray from '../styles.css';
}

.icon-left {
  margin-right: 0.25rem;
}