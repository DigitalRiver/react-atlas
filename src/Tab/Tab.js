import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./Tab.css";

/**
 * Tab component
 */
export class Tab extends React.PureComponent {
  constructor(props) {
    super(props);

    // Initial state
    this.state = {
      selected: this.props.selected,
      disabled: this.props.disabled
    };
  }

  UNSAFE_componentWillReceiveProps = nextProps => {
    if (nextProps.selected !== this.props.selected) {
      this.setState({ selected: nextProps.selected });
    }
  };

  _handleClick = event => {
    let selected = this.props.tabIndex;

    if (!this.state.disabled) {
      this.props.setSelectedTab(selected, event);
    }
  };

  render() {
    const { className, children, vertical, icon, style } = this.props;

    let tabClasses = cx({
      tab: !vertical,
      vTab: vertical,
      selected: this.state.selected && !vertical,
      vSelected: this.state.selected && vertical,
      disabled: this.state.disabled
    });

    return (
      <li
        className={cx(className)}
        styleName={tabClasses}
        style={style}
        onClick={this._handleClick}
      >
        {icon ? <i className={cx(icon, "ra_Tab__icon-left")} /> : null}
        {children}
      </li>
    );
  }
}

Tab.propTypes = {
  /**
   * Child components or elements to be displayed as the title on Tab.
   */
  children: PropTypes.node,

  /**
   * An object, array, or string of CSS classes to apply to Tab.
   */
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * When true, Tab sill be disabled.
   */
  disabled: PropTypes.bool,

  /**
   * The CSS class name for the icon that will be displayed on Tab.
   */
  icon: PropTypes.string,

  /**
   * When true, Tab will be selected.
   */
  selected: PropTypes.bool,

  /**
   * Function to be executed when a Tab is selected.
   */
  setSelectedTab: PropTypes.func,

  /**
   * Pass inline styling here.
   */
  style: PropTypes.object,

  /**
   * Tab index.
   */
  tabIndex: PropTypes.number,

  /**
   * When true, Tabs will display vertically.
   */
  vertical: PropTypes.bool
};

Tab.defaultProps = {
  selected: false,
  disabled: false
};

export default CSSModules(Tab, styles, { allowMultiple: true });
