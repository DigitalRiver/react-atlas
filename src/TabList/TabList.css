@import '../styles/variables.css';

.tabList {
  composes: marg-0 from '../styles.css';
  composes: pad-0 from '../styles.css';
  border-bottom: 1px solid var(--gray);
}

.vertical {
  flex-grow: 1;
  border-bottom: none;
  border-right: 1px solid var(--gray);
}