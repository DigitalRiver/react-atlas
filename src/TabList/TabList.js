import React, { cloneElement } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import CSSModules from "react-css-modules";
import styles from "./TabList.css";

/**
 * TabList component
 */
export class TabList extends React.PureComponent {
  constructor(props) {
    super(props);

    // Initial state
    this.state = {
      selectedTab: this.props.selectedTab || 0
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.selectedTab !== this.props.selectedTab) {
      this.setState({ selectedTab: nextProps.selectedTab });
    }
  }

  _setSelectedTab = (index, event) => {
    this.setState({ selectedTab: index });
    this.props.setSelectedIndex(index, event);
  };

  render() {
    const { className, children, vertical, style } = this.props;

    const tabs = React.Children.map(children, (child, index) => {
      child = cloneElement(child, {
        tabIndex: index,
        selected: this.state.selectedTab === index,
        setSelectedTab: this._setSelectedTab,
        vertical: vertical
      });

      return child;
    });

    let tabListClasses = cx(
      {
        vertical
      },
      "tabList"
    );

    return (
      <ul className={cx(className)} styleName={tabListClasses} style={style}>
        {tabs}
      </ul>
    );
  }
}

TabList.propTypes = {
  /**
   * Will be automatically set when bordered prop is passed to Tabs component.
   *
   * @ignore
   */
  bordered: PropTypes.bool,

  /**
   * An object, array, or string of CSS classes to apply to TabPanel.
   */
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * Tab components that will be displayed in TabList.
   */
  children: PropTypes.node.isRequired,

  /**
   * Selected tab index (default 0 - first tab).
   */
  selectedTab: PropTypes.number,

  /**
   * Handler to execute when a tab is selected, in Tabs component.
   */
  setSelectedIndex: PropTypes.func,

  /**
   * Pass inline styling here.
   */
  style: PropTypes.object,

  /**
   * Will be automatically set when vertical prop is passed to Tabs component.
   */
  vertical: PropTypes.bool
};

export default CSSModules(TabList, styles, { allowMultiple: true });
