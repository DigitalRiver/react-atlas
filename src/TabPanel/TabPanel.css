@import '../styles/variables.css';

.tabPanel {
  display: none;
}

.selected {
  display: block;
}

.bordered {
  border: 1px solid var(--gray);
  border-top: none;
  padding: 1rem;
}

.vertical {
  flex-grow: 8;
  border: 1px solid var(--gray);
  border-left: none;
  padding: 1rem;
}