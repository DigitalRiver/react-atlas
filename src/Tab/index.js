import Tab from "./Tab.js";
export { Tab };
