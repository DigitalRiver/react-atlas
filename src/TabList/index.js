import TabList from "./TabList.js";
export { TabList };
