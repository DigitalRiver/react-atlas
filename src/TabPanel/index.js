import TabPanel from "./TabPanel.js";
export { TabPanel };
