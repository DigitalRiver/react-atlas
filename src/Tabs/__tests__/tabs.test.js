import React from "react";
import { mount } from "enzyme";
import { Tabs } from "../index";
import { TabList } from "../../TabList/index";
import { Tab } from "../../Tab/index";
import { TabPanel } from "../../TabPanel/index";
import renderer from "react-test-renderer";

describe("Test dropdown component", () => {
  it("Test render correctly", () => {
    const comp = (
      <Tabs>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
          <Tab>Tab 3</Tab>
          <Tab>Tab 4</Tab>
          <Tab>Tab 5</Tab>
        </TabList>
        <TabPanel>
          <h2>Content 1</h2>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin
            laoreet lorem sit amet finibus rutrum. Vestibulum ante ipsum primis
            in faucibus orci luctus et ultrices posuere cubilia Curae;
            Pellentesque habitant morbi tristique senectus et netus et malesuada{" "}
            <a href="http://lipsum.com">fames ac turpis egestas</a> et siem.
          </p>
        </TabPanel>
        <TabPanel>
          <h2>Content 2</h2>
          <p>Tab content here</p>
        </TabPanel>
        <TabPanel>
          <h2>Content 3</h2>
          <p>Tab content here</p>
        </TabPanel>
        <TabPanel>
          <h2>Content 4</h2>
          <p>Tab content here</p>
        </TabPanel>
        <TabPanel>
          <h2>Content 5</h2>
          <p>Tab content here</p>
        </TabPanel>
      </Tabs>
    );
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders border", () => {
    const tabs = mount(
      <Tabs bordered>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
        </TabList>
        <TabPanel>
          <h2>Content 1</h2>
          <p>
            Lorem ipsum dolor sit amet{" "}
            <a href="http://lipsum.com">fames ac turpis egestas</a> et siem.
          </p>
        </TabPanel>
        <TabPanel>
          <h2>Content 2</h2>
          <p>Tab content here</p>
        </TabPanel>
      </Tabs>
    );

    expect(tabs.find(".bordered").exists()).toBe(true);
  });

  it("sets defaults", () => {
    const tabs = mount(
      <Tabs>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
        </TabList>
        <TabPanel>
          <h2>Content 1</h2>
          <p>
            Lorem ipsum dolor sit amet{" "}
            <a href="http://lipsum.com">fames ac turpis egestas</a> et siem.
          </p>
        </TabPanel>
        <TabPanel>
          <h2>Content 2</h2>
          <p>Tab content here</p>
        </TabPanel>
      </Tabs>
    );

    expect(tabs.find(".bordered").exists()).not.toBe(true);
    expect(tabs.find(".vertical").exists()).not.toBe(true);
    expect(tabs.state().selectedIndex).toBe(0);
  });

  it("applies className", () => {
    const tabs = mount(
      <Tabs className="myClass">
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
        </TabList>
        <TabPanel>
          <h2>Content 1</h2>
          <p>
            Lorem ipsum dolor sit amet{" "}
            <a href="http://lipsum.com">fames ac turpis egestas</a> et siem.
          </p>
        </TabPanel>
        <TabPanel>
          <h2>Content 2</h2>
          <p>Tab content here</p>
        </TabPanel>
      </Tabs>
    );

    expect(tabs.find(".myClass").exists()).toBe(true);
  });

  it("handles onSelect", () => {
    const handleOnSelect = jest.fn();
    const tabs = mount(
      <Tabs onSelect={handleOnSelect}>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
        </TabList>
        <TabPanel>
          <h2>Content 1</h2>
          <p>
            Lorem ipsum dolor sit amet{" "}
            <a href="http://lipsum.com">fames ac turpis egestas</a> et siem.
          </p>
        </TabPanel>
        <TabPanel>
          <h2>Content 2</h2>
          <p>Tab content here</p>
        </TabPanel>
      </Tabs>
    );

    tabs
      .find("Tab")
      .first()
      .simulate("click");
    expect(handleOnSelect).toBeCalledWith(0, expect.any(Object));
    expect(tabs.state().selectedIndex).toBe(0);
    tabs
      .find("Tab")
      .last()
      .simulate("click");
    expect(handleOnSelect).toBeCalledWith(1, expect.any(Object));
    expect(tabs.state().selectedIndex).toBe(1);
  });

  it("does not run onSelect when onSelect prop not present", () => {
    const handleOnSelect = jest.fn();
    const tabs = mount(
      <Tabs>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
        </TabList>
        <TabPanel>
          <h2>Content 1</h2>
          <p>
            Lorem ipsum dolor sit amet{" "}
            <a href="http://lipsum.com">fames ac turpis egestas</a> et siem.
          </p>
        </TabPanel>
        <TabPanel>
          <h2>Content 2</h2>
          <p>Tab content here</p>
        </TabPanel>
      </Tabs>
    );

    tabs
      .find("Tab")
      .first()
      .simulate("click");
    expect(handleOnSelect).not.toBeCalled();
  });

  it("sets selectedIndex to 0 by default", () => {
    const handleOnSelect = jest.fn();
    const tabs = mount(
      <Tabs onSelect={handleOnSelect}>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
        </TabList>
        <TabPanel>
          <h2>Content 1</h2>
          <p>
            Lorem ipsum dolor sit amet{" "}
            <a href="http://lipsum.com">fames ac turpis egestas</a> et siem.
          </p>
        </TabPanel>
        <TabPanel>
          <h2>Content 2</h2>
          <p>Tab content here</p>
        </TabPanel>
      </Tabs>
    );

    expect(tabs.state().selectedIndex).toBe(0);
  });

  it("CWRP updates selectedIndex state", () => {
    const tabs = mount(
      <Tabs>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
        </TabList>
        <TabPanel>
          <h2>Content 1</h2>
          <p>
            Lorem ipsum dolor sit amet{" "}
            <a href="http://lipsum.com">fames ac turpis egestas</a> et siem.
          </p>
        </TabPanel>
        <TabPanel>
          <h2>Content 2</h2>
          <p>Tab content here</p>
        </TabPanel>
      </Tabs>
    );

    expect(tabs.state().selectedIndex).toBe(0);
    tabs.setProps({ selectedIndex: 1 });
    expect(tabs.state().selectedIndex).toBe(1);
  });

  it("applies style", () => {
    const tabs = mount(
      <Tabs style={{ color: "red" }}>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
        </TabList>
        <TabPanel>
          <h2>Content 1</h2>
          <p>
            Lorem ipsum dolor sit amet{" "}
            <a href="http://lipsum.com">fames ac turpis egestas</a> et siem.
          </p>
        </TabPanel>
        <TabPanel>
          <h2>Content 2</h2>
          <p>Tab content here</p>
        </TabPanel>
      </Tabs>
    );

    expect(tabs.find(".tabs").props().style).toEqual({ color: "red" });
  });

  it("applies vertical", () => {
    const tabs = mount(
      <Tabs vertical>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
        </TabList>
        <TabPanel>
          <h2>Content 1</h2>
          <p>
            Lorem ipsum dolor sit amet{" "}
            <a href="http://lipsum.com">fames ac turpis egestas</a> et siem.
          </p>
        </TabPanel>
        <TabPanel>
          <h2>Content 2</h2>
          <p>Tab content here</p>
        </TabPanel>
      </Tabs>
    );

    expect(tabs.find(".vertical.tabs").exists()).toBe(true);
  });
});
