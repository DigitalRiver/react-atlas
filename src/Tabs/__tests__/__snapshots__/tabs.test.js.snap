// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test dropdown component Test render correctly 1`] = `
<div
  className="tabs"
  style={undefined}
>
  <ul
    className="tabList"
    style={undefined}
  >
    <li
      className="tab selected"
      onClick={[Function]}
      style={undefined}
    >
      Tab 1
    </li>
    <li
      className="tab"
      onClick={[Function]}
      style={undefined}
    >
      Tab 2
    </li>
    <li
      className="tab"
      onClick={[Function]}
      style={undefined}
    >
      Tab 3
    </li>
    <li
      className="tab"
      onClick={[Function]}
      style={undefined}
    >
      Tab 4
    </li>
    <li
      className="tab"
      onClick={[Function]}
      style={undefined}
    >
      Tab 5
    </li>
  </ul>
  <div
    className="selected tabPanel"
    style={undefined}
  >
    <h2>
      Content 1
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin laoreet lorem sit amet finibus rutrum. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Pellentesque habitant morbi tristique senectus et netus et malesuada
       
      <a
        href="http://lipsum.com"
      >
        fames ac turpis egestas
      </a>
       et siem.
    </p>
  </div>
  <div
    className="tabPanel"
    style={undefined}
  >
    <h2>
      Content 2
    </h2>
    <p>
      Tab content here
    </p>
  </div>
  <div
    className="tabPanel"
    style={undefined}
  >
    <h2>
      Content 3
    </h2>
    <p>
      Tab content here
    </p>
  </div>
  <div
    className="tabPanel"
    style={undefined}
  >
    <h2>
      Content 4
    </h2>
    <p>
      Tab content here
    </p>
  </div>
  <div
    className="tabPanel"
    style={undefined}
  >
    <h2>
      Content 5
    </h2>
    <p>
      Tab content here
    </p>
  </div>
</div>
`;
